### PR TITLE
split API from implementation

### PR DESCRIPTION
**Pull request by aserkes**
split API from implementation
Signed-off-by: aserkes <andrii.serkes@oracle.com>

### CODE REVIEW

**Review by lukasj — COMMENTED** [review]
In general - I like this! I will have some comments as I go through this. First are:
* new files should have only 2021 in cp year
* implementation should also contain module-info
* there needs to be better (and/or properly documented) way to handle missing implementation - `ServiceLoader.load(MailcapRegistryProvider.class).iterator().next().getDefault()` may throw an exception if there is no implementation available

**Review by lukasj — CHANGES_REQUESTED** [review]
* new interfaces in `jakarta.activation.spi` package require javadoc
* consider renaming `getDefault` methods to `getEmpty` (or something more appropriate)

**Review by lukasj — APPROVED** [review]
I think this is OK. We can fine-tune it later. Just rebase it, so it can be merged. Thanks!

File: activation/src/main/java/jakarta/activation/MimeTypeRegistry.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+     *
+     * @param    file_ext    the file extension
+     * @return the MIME type string
+     */
+    default String getMIMETypeString(String file_ext) {
+        MimeTypeEntry entry = this.getMimeTypeEntry(file_ext);
+
+        if (entry != null)
+            return entry.getMIMEType();
+        else
```
Comment (lukasj) [inline activation/src/main/java/jakarta/activation/MimeTypeRegistry.java:44]: `else` is redundant here

File: activation/src/main/java/jakarta/activation/MimetypesFileTypeMap.java [hunk 2]
```diff
@@ -213,7 +220,12 @@ private void loadAllResources(Vector v, String name) {
 			if (LogSupport.isLoggable())
 			    LogSupport.log("MimetypesFileTypeMap: can't load " +
 						url, sex);
-		    } finally {
+		    } catch (NoSuchElementException | ServiceConfigurationError e) {
+				if (LogSupport.isLoggable()) {
```
Comment (lukasj) [inline activation/src/main/java/jakarta/activation/MimetypesFileTypeMap.java:224]: the spaces here look like `tabs`. Make sure real space characters are used (no `tabs`)

File: activation/src/main/java/jakarta/activation/MimetypesFileTypeMap.java [hunk 3]
```diff
@@ -238,13 +250,18 @@ private void loadAllResources(Vector v, String name) {
...
-    private MimeTypeFile loadFile(String name) {
-	MimeTypeFile mtf = null;
+    private MimeTypeRegistry loadFile(String name) {
+	MimeTypeRegistry mtf = null;
 
 	try {
-	    mtf = new MimeTypeFile(name);
+	    mtf = ServiceLoader.load(MimeTypeRegistryProvider.class).iterator().next().getByFileName(name);
 	} catch (IOException e) {
 	    //	e.printStackTrace();
```
Comment (lukasj) [inline activation/src/main/java/jakarta/activation/MimetypesFileTypeMap.java:259]: do not swallow exceptions. Either remove the catch or move it to the next one so it is at least logged

File: activation/src/main/java/jakarta/activation/spi/package.html [hunk 4]
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
+<HTML>
```
Comment (lukasj) [inline activation/src/main/java/jakarta/activation/spi/package.html:2]: it's better to use `package-info.java` files instead of html

**Comment by aserkes** [comment]
All fixes are done. Code is ready to merge.

**Pull request merged**